src/audio/google/google_ctc_audio_processing_ipc4.c

```diff
@@ -25,16 +25,26 @@ int ctc_set_config(struct processing_module *mod, uint32_t param_id,
 		   size_t fragment_size, uint8_t *response,
 		   size_t response_size)
 {
+	struct sof_ipc4_control_msg_payload *ctl = (struct sof_ipc4_control_msg_payload *)fragment;
 	struct google_ctc_audio_processing_comp_data *cd = module_get_private_data(mod);
 	int ret;
 
 	switch (param_id) {
 	case 0:
 		break;
 	case SOF_IPC4_SWITCH_CONTROL_PARAM_ID:
+		if (ctl->id == 0 && ctl->num_elems == 1) {
+			cd->enabled = ctl->chanv[0].value;
+			comp_info(mod->dev, "ctc_set_config(): enabled = %d", cd->enabled);
+		} else {
+			comp_err(mod->dev, "ctc_set_config(): Illegal control id = %d, num_elems = %d",
+				 ctl->id, ctl->num_elems);
+			return -EINVAL;
+		}
+		return 0;
```

Comment:
ditto to @kv2019i comments. can this be cleaned to single branch?
```
		if (ctl->id == 0 && ctl->num_elems == 1) {
			cd->enabled = ctl->chanv[0].value;
			comp_info(mod->dev, "ctc_set_config(): enabled = %d", cd->enabled);
                        return 0;
		} 
		comp_err(mod->dev, "ctc_set_config(): Illegal control id = %d, num_elems = %d",
				 ctl->id, ctl->num_elems);
		return -EINVAL;
```

Reply:
done.